app/ide-desktop/lib/client/src/index.ts

```diff
@@ -74,6 +140,44 @@ class App {
         }
     }
 
+    processArguments() {
+        // We parse only "client arguments", so we don't have to worry about the Electron-Dev vs
+        // Electron-Proper distinction.
+        let openedFile = attemptingToOpenFile(paths.clientArguments)
+        // If we are opening a file (i.e. we weere spawned with just a path of the file to open as
+        // the argument), it means that effectively we don't have any non-standard arguments.
+        // We just need to let caller know that we are opening a file.
+        let argsToParse = openedFile ? [] : paths.clientArguments
+        return { ...configParser.parseArgs(argsToParse), openedFile }
+    }
+
+    /** Handle the case where IDE is invoked with a file to open.
+     *
+     * Imports project if necessary. Returns the ID of the project to open.
+     *
```

Comment:
Unnecessary nl. 3 and 5 line are additional docs, and they should be glued together. The first nl makes sense, as the first line is a synopsis.